lectures/olg.md

```diff
@@ -155,5 +187,288 @@ $k_{t+1} = k_t = k^*$, i.e,
+This is a form of global stability.
+
+
+The next figure shows three time paths for capital, from
+three distinct initial conditions, under the parameterization listed above.
+
+At this parameterization, $k^* \approx 0.161$.
+
+Let's define the constants and three distinct intital conditions
+
+```{code-cell} ipython3
+ts_length = 10
+x0 = np.array([0.001, 1.2, 2.6])
+```
+
+```{code-cell} ipython3
+def simulate_ts(olg, x0_values, ts_length):
+
+    k_star = optimize.newton(solve_for_k_star, 0.2, args=(olg,))
+    fig, ax = plt.subplots(figsize=(10, 5))
+
+    ts = np.zeros(ts_length)
+
+    # simulate and plot time series
+    for x_init in x0_values:
+        ts[0] = x_init
+        for t in range(1, ts_length):
+            ts[t] = k_next(olg, ts[t-1])
+        ax.plot(np.arange(ts_length), ts, '-o', ms=4, alpha=0.6,
+                label=r'$k_0=%g$' %x_init)
+    ax.plot(np.arange(ts_length), np.full(ts_length,k_star),
+            alpha=0.6, color='red', label=r'$k^*$')
+    ax.legend(fontsize=10)
+
+    ax.set_xlabel(r'$t$', fontsize=14)
+    ax.set_ylabel(r'$k_t$', fontsize=14)
+
+    plt.show()
+```
+
+```{code-cell} ipython3
+simulate_ts(olg, x0, ts_length)
+```
+
++++ {"tags": []}
+
+## Exercises
+
+```{exercise}
+:label: olg_ex1
+
+Replace the utility function $u(c)$ in equation {eq}`eq_crra`  with a quasilinear form $u(c)=c + c^{\theta}$.
+
+Now we don't have an analytical solution. 
+
+Try to compute the time path capital $\{k_t\}$ in this case.
+```
+
++++
+
+From {eq}`max_sav_olg`, the first-order condition for a maximum can be written in the
+familiar form of the consumption Euler equation given by
+
+$$
+    u'(c^1_t) = \beta R_{t+1} u'(c^2_{t+1})
+$$
+
+Using {eq}`c_2_olg`, we have
+
+$$
+    u'(c^1_t) = \beta R_{t+1} u'(R_{t+1}(w_t - c^1_t))
+$$
+
+Substituting using {eq}`R_func`, and {eq}`w_func`
+
+$$
+    u'(c^1_t) = \beta \alpha k^{\alpha-1}_{t+1} u'(\alpha k^{\alpha-1}_{t+1}((1 - \alpha)k^{\alpha}_t - c^1_t))
+$$
+
+Since $u$ is quasilinear, in order to solve for $c^1_t$, we will need to use [scipy.optimize.newton](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.newton.html#scipy.optimize.newton).
+
+From {eq}`k_dyms_crra`,
+
+$$
+    k_{t+1} = \frac{s(w_t, R_{t+1})}{1 + n} = \frac{w_t - c^1_t}{1+n}
+$$
+
++++
+
+Let's define the functions to solve the above system.
+
+```{code-cell} ipython3
+def u_prime(c, θ):
+    return 1 + θ * c ** (θ - 1)
+
+def solve_for_c(x, θ, model, k_t, k_t_1):
+    l = u_prime(x, θ)
+    R = model.α * k_t_1**(model.α-1)
+    r = model.β * R
+    y = R * ((1 - model.α) * k_t**model.α - x)
+    r = r * u_prime(y, θ)
+    return l - r
+```
+
+```{code-cell} ipython3
+def solve_for_k(x, θ, model, k_t):
+    c = optimize.newton(solve_for_c, k_t, args=(θ, model, k_t, x))
+    num = (1 - model.α) * k_t**model.α - c
+    den = (1 + model.n)
+    return x - num/den
+
+def k_next(model, θ, k_t):
+    return optimize.newton(solve_for_k, k_t, args=(θ, model, k_t), maxiter=200)
+```
+
```

Comment:
I'm trying to solve the quasilinear utility exercise here. I'm getting an error that `c` was not able to converge. Maybe the problem is getting more complex as to find `k`, I'm using newton's and to find the value of `c`, I'm again using newton's method. This might be the issue here. @jstac Can you please have a look and please let me know if you think there's a workaround to this?